syntax = "proto3";

message Settlement {
  message Economy {
    int32 id = 1;
    string type = 2;
    string sub_type = 3;
  }

  message Metadata {
    string type = 1;
    float distance = 2;
    string allegiance = 3;
    int32 controlling_faction = 4;
    string government = 5;
  }

  message Services {
    bool market = 1;
    bool shipyard = 2;
    bool outfitting = 3;
    repeated string other = 4;
  }

  int32 id = 1;
  int32 system_id = 2;
  string name = 3;
  Metadata metadata = 4;
  Economy economy = 5;
  Services services = 6;
  int32 updated = 7;
}
